tracing-attributes/src/lib.rs

```diff
@@ -468,8 +468,7 @@ fn gen_body(
         quote_spanned!(block.span()=>
             let __tracing_attr_span = #span;
             let __tracing_attr_guard = __tracing_attr_span.enter();
-            let f = move || #return_type { #block };
-            match f() {
+            match move || #return_type { #block }() {
```

Comment:
hmm, this doesn't seem to compile --- it interprets this as a `()` value, rather than calling the function. i think we need to write this:
```suggestion
            match (move || #return_type { #block })() {
```